parcels/field.py

```diff
@@ -68,13 +68,14 @@ def from_netcdf(cls, name, varname, datasets, **kwargs):
         """
         if not isinstance(datasets, Iterable):
             datasets = [datasets]
-        lon = datasets[0]['nav_lon'][0, :]
-        lat = datasets[0]['nav_lat'][:, 0]
+        lon = datasets[0].variables[dimensions["long"]]
```

Comment:
- Ah yes, apologies, I completely forgot about the tests, I should have
   run them first. They check for PEP8 right? Getting used to the workflow.
   - If you want to create a directory structure for tests and datafiles
   that fits those requirements, I can change the locations of files and
   default settings in the test file easily enough.
   - Yes we discussed the superclass framework- I thought I would make my
   first pull request based on this standalone class for the moment. In a few
   weeks I should *hopefully* have yet another type of field data from our
   colleagues in Toulouse (habitat-maps), so perhaps it is worth waiting to
   see what these look like before we build a higher level class?
   - 'Long' was my instinctive choice for longitude dimension naming- it
   was only afterwards that I saw that the field class uses 'lon', sorry.
   - The previous plotParticles was already hardcoded to NEMO so it was
   already breaking for files outside that convention. Agree on making it
   generic.
   - Acceptance criteria... Perhaps that the resulting grid actually
   contains some data? Animating the flowfields as background images in
   plotParticles would be really great as well.


On 15 February 2016 at 12:07, Michael Lange <notifications@github.com>
wrote:

> In parcels/field.py
> <https://github.com/OceanPARCELS/parcels/pull/35#discussion_r52891366>:
>
> > @@ -68,13 +68,14 @@ def from_netcdf(cls, name, varname, datasets, **kwargs):
> >          """
> >          if not isinstance(datasets, Iterable):
> >              datasets = [datasets]
> > -        lon = datasets[0]['nav_lon'][0, :]
> > -        lat = datasets[0]['nav_lat'][:, 0]
> > +        lon = datasets[0].variables[dimensions["long"]]
>
> Is there any particular reason why we switch from lon to long here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OceanPARCELS/parcels/pull/35/files#r52891366>.
>